jax/_src/lax/lax.py

```diff
@@ -3390,8 +3390,8 @@ def _reduce_lower(ctx, *values, computation, jaxpr, consts, dimensions):
   reducer = op.regions[0].blocks.append(*(ir_types + ir_types))
   with ir.InsertionPoint(reducer):
     reducer_ctx = ctx.module_context.replace(name_stack=util.new_name_stack())
-    out_nodes = mlir.jaxpr_subcomp(reducer_ctx, jaxpr, consts,
-                                   *([a] for a in reducer.arguments))
+    out_nodes, _ = mlir.jaxpr_subcomp(reducer_ctx, jaxpr, mlir.TokenSet(), consts,
```

Comment:
Should we also be raising a NotImplementedError in this case if the jaxpr has effects? (Basically, whenever we have an underscore dropping the token coming out of `mlir.jaxpr_subcomp`, it seems like we need a corresponding error if the jaxpr is effectful.)

Reply:
Maybe we could even assert that the token going in is same object as the token coming out. That could be an interesting check!

Reply:
I'll add in the `NotImplementedError`s.

There probably won't be a token going in because it will be filtered at an earlier call site.